src/test/java/org/seqdoop/hadoop_bam/TestCRAMOutputFormat.java

```diff
@@ -0,0 +1,156 @@
+        final SAMFileHeader header = samReader.getFileHeader();
+
+        final KeyIgnoringCRAMOutputFormat<NullWritable> cramOut = new KeyIgnoringCRAMOutputFormat<NullWritable>();
+        cramOut.setWriteHeader(false);
+        cramOut.setSAMHeader(header);
+
+        RecordWriter<NullWritable, SAMRecordWritable> rw = cramOut.getRecordWriter(taskAttemptContext, out);
+        int expectedCount = 0;
+        for (final SAMRecord r : samReader) {
+            final SAMRecordWritable samRW = new SAMRecordWritable();
+            samRW.set(r);
+            rw.write(null, samRW);
+            expectedCount++;
+        }
+        samReader.close();
+        rw.close(taskAttemptContext);
+
+        // now verify the container stream
+        final ReferenceSource refSource = new ReferenceSource(Paths.get(reference));
+        final int actualCount = verifyContainerStream(new File(cramOutFileName), header, refSource);
+
+        assertEquals(expectedCount, actualCount);
+    }
+
+
+    @Test
+    public void testCRAMOutputFormatMapReduce() throws Exception {
+        conf = new Configuration();
+        conf.set(CRAMInputFormat.REFERENCE_SOURCE_PATH_PROPERTY, "file://" + reference);
+
+        final FileSystem fileSystem = FileSystem.get(conf);
+        final Path inputPath = new Path(input);
+        final Path outputPath = fileSystem.makeQualified(new Path("target/out"));
+        fileSystem.delete(outputPath, true);
+
+        final Job job = Job.getInstance(conf);
+        FileInputFormat.setInputPaths(job, inputPath);
+
+        job.setInputFormatClass(CRAMInputFormat.class);
+        job.setMapOutputKeyClass(LongWritable.class);
+        job.setMapOutputValueClass(SAMRecordWritable.class);
+
+        job.setOutputFormatClass(KeyIgnoringCRAMOutputFormat.class);
+        job.setOutputKeyClass(LongWritable.class);
+        job.setOutputValueClass(SAMRecordWritable.class);
+
+        job.setNumReduceTasks(0);
+        FileOutputFormat.setOutputPath(job, outputPath);
+
+        final boolean success = job.waitForCompletion(true);
+        assertTrue(success);
+
+        // now verify
+        // fetch the SAMFile header from the original input to get eh expected count
+        final ReferenceSource refSource = new ReferenceSource(Paths.get(reference));
+        final CRAMFileReader cramReader = new CRAMFileReader(new File(input), (File)null, refSource);
+        final SAMFileHeader header = cramReader.getFileHeader();
+        final Iterator<SAMRecord> it = cramReader.getIterator();
+        int expectedCount = 0;
+        while (it.hasNext()) {
+            final SAMRecord rec = it.next();
+            expectedCount++;
+        }
+        cramReader.close();
+
+        final File containerStreamFile = new File(new File(outputPath.toUri()), "part-m-00000");
+        final int actualCount = verifyContainerStream(containerStreamFile, header, refSource);
+        assertEquals(expectedCount, actualCount);
+    }
+
```

Comment:
Can you add a test that the CRAMInputFormat can read the output from CRAMOutputFormat?